src/Microsoft.AspNetCore.Authentication.Cookies/CookieExtensions.cs

```diff
@@ -15,7 +19,62 @@ public static class CookieExtensions
         public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, Action<CookieAuthenticationOptions> configureOptions) =>
             services.AddCookieAuthentication(CookieAuthenticationDefaults.AuthenticationScheme, configureOptions);
 
-        public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, string authenticationScheme, Action<CookieAuthenticationOptions> configureOptions) =>
-            services.AddScheme<CookieAuthenticationOptions, CookieAuthenticationHandler>(authenticationScheme, configureOptions);
+        public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, string authenticationScheme, Action<CookieAuthenticationOptions> configureOptions)
+        {
+            // Makes sure that DataProtectionProvider is always set.
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IInitializeOptions<CookieAuthenticationOptions>, EnsureDataProtection>());
+            services.Initialize<CookieAuthenticationOptions>(authenticationScheme, options =>
```

Comment:
Hrm, do we want to allow people to replace/customize this behavior?  They already can via adding their own Initializer after, or Configuring it before.  I guess it doesn't hurt to leave it public, altho on an unrelated note, all of our handlers are still internal which probably should change now that they actually can be extended